org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelCodeLensParticipant.java

```diff
@@ -62,9 +67,60 @@
  */
 public class ContentModelCodeLensParticipant implements ICodeLensParticipant {
 
+	private final ContentModelManager contentModelManager;
+
+	public ContentModelCodeLensParticipant(ContentModelManager contentModelManager) {
+		this.contentModelManager = contentModelManager;
+	}
+
 	@Override
 	public void doCodeLens(ICodeLensRequest request, List<CodeLens> lenses, CancelChecker cancelChecker) {
-		if (!canSupport(request)) {
+		// List of referenced schema/grammar
+		createReferencedGrammarLenses(request, lenses);
+		// "Bind to grammar/schema..."
+		createBindToGrammarSchemaLenses(request, lenses);
+	}
+
+	private void createReferencedGrammarLenses(ICodeLensRequest request, List<CodeLens> lenses) {
+		if (!canSupportOpenUri(request)) {
+			return;
+		}
+		
+		DOMDocument document = request.getDocument();
+		if (!document.hasGrammar()) {
+			return;
+		}
+		Range range = XMLPositionUtility.createRange(document);

```

Comment:
Seems like the range for a CodeLens is expected to only span one line (See the definition of the `CodeLens` interface): https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_codeLens

Reply:
Fixed, now the start and the end position have the same value.